lib/flags/local_flags.js

```diff
@@ -0,0 +1,412 @@
+/**
+ * Local Feature Flags Provider
+ * Evaluates feature flags client-side using locally cached definitions
+ */
+
+/**
+ *  @typedef {import('./types').SelectedVariant} SelectedVariant
+ *  @typedef {import('./types').FlagContext} FlagContext
+ *  @typedef {import('./types').LocalFlagsConfig} LocalFlagsConfig
+ *  @typedef {import('./types').ExperimentationFlag} ExperimentationFlag
+ *  @typedef {import('./types').LocalFlagsResponse} LocalFlagsResponse
+ * */
+
+const https = require('https');
+const packageInfo = require('../../package.json');
+const {
+    EXPOSURE_EVENT,
+    REQUEST_HEADERS,
+    prepareCommonQueryParams,
+    normalizedHash,
+} = require('./utils');
+
+class LocalFeatureFlagsProvider {
+    /**
+     * @param {string} token - Mixpanel project token
+     * @param {LocalFlagsConfig} config - Local flags configuration
+     * @param {Function} tracker - Function to track events (signature: track(distinct_id, event, properties, callback))
+     * @param {CustomLogger} logger - Logger
+     */
+    constructor(token, config, tracker, logger) {
+        this.token = token;
+        this.config = {
+            api_host: 'api.mixpanel.com',
+            request_timeout_in_seconds: 10,
+            enable_polling: true,
+            polling_interval_in_seconds: 60,
+            ...config,
+        };
+        this.tracker = tracker;
+        this.logger = logger;
+
+        this.flagDefinitions = new Map();
+        this.pollingInterval = null;
+    }
+
+    /**
+     * Start polling for flag definitions.
+     * Fetches immediately and then at regular intervals if polling is enabled
+     * @returns {Promise<void>}
+     */
+    async startPollingForDefinitions() {
+        try {
+            await this._fetchFlagDefinitions();
+
+            if (this.config.enable_polling && !this.pollingInterval) {
+                this.pollingInterval = setInterval(async () => {
+                    try {
+                        await this._fetchFlagDefinitions();
+                    } catch (err) {
+                        this.logger?.error(`Error polling for flag definition: ${err.message}`);
+                    }
+                }, this.config.polling_interval_in_seconds * 1000);
+            }
+        } catch (err) {
+            this.logger?.error(`Initial flag definitions fetch failed: ${err.message}`);
+        }
+    }
+
+    /**
+     * Stop polling for flag definitions
+     */
+    stopPollingForDefinitions() {
+        if (this.pollingInterval) {
+            clearInterval(this.pollingInterval);
+            this.pollingInterval = null;
+        } else {
+            this.logger?.warn('stopPollingForDefinitions called but polling was not active');
+        }
+    }
+
+    /**
+     * Check if a feature flag is enabled
```

Comment:
Might wanna add a note that this is strictly checking for `true` and not doing a thruthy check for the js folks